### PR TITLE
Update the dependencies so the application runs on Apple Silicon

### PR DESCRIPTION
**Pull request by hegyibalint**
Our spring-boot (<2.7) didn't support Apple Silicon.

I took this chance to update most of our versions and see if things still work.
Also, as I've updated spotless, I've applied most of the changes it offered by running `spotlessKotlinApply`.

### CODE REVIEW

File: build-platform/build.gradle.kts [hunk 1]
```diff
@@ -7,16 +7,9 @@ javaPlatform {
...
 
     constraints {
-        api("org.apache.logging.log4j:log4j-api:2.16.0")
-        api("org.apache.logging.log4j:log4j-to-slf4j:2.16.0")
-        api("org.flywaydb:flyway-core:7.7.1")
-        api("org.jetbrains.teamcity:teamcity-rest-client:1.18.1")
-        api("org.junit.jupiter:junit-jupiter-params:5.9.1")
-        api("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.2") {
-            because("used by ge-export:test")
-        }
```
Comment (hegyibalint) [inline build-platform/build.gradle.kts:20]: Except the TC rest client, all of these are coming from the `spring-boot-dependencies`.

I always let it drive the dependencies, as it provides maximum compatibility with spring boot.

File: ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt [hunk 2]
```diff
@@ -84,7 +84,7 @@ object TaskSummaryExtractor : SingleEventExtractor<TaskSummary>("TaskFinished")
             events
                 .filter { it.data?.stringProperty("outcome") != null }
                 .groupBy { it.data?.stringProperty("outcome") }
-                .mapKeys { TaskOutcome.valueOf(it.key!!.toUpperCase()) }
```
Comment (hegyibalint) [inline ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt:87]: `toUpperCase` was deprecated in later Kotlin versions.

File: ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt [hunk 3]
```diff
@@ -117,7 +117,7 @@ object ExecutedTestTasks : Extractor<List<String>>(listOf("TestStarted", "TestFi
         return events
             .getOrDefault(LongTestClassExtractor.eventTypes[1], emptyList())
             .filter {
-                it.data?.stringProperty("outcome")?.toUpperCase() in listOf("SUCCESS", "FAILED") &&
```
Comment (hegyibalint) [inline ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt:120]: `toUpperCase` was deprecated in later Kotlin versions.

File: build-platform/build.gradle.kts (hunk 1 above)
Comment (wolfs) [reply build-platform/build.gradle.kts:20]: Interesting, I think those haven't been part of the platform in 2.5.

File: application/src/main/resources/application.properties [hunk 4]
```diff
@@ -4,5 +4,5 @@ gradle.enterprise.apiToken=${GRADLE_ENTERPRISE_EXPORT_API_TOKEN}
 teamcity.api.token=${TEAMCITY_API_TOKEN}
 spring.datasource.url=${JDBC_DATABASE_URL:jdbc:postgresql://localhost:5432/btdevprod}
 spring.datasource.username=${JDBC_DATABASE_USERNAME:btdevprod}
-spring.datasource.password=${JDBC_DATABASE_PASSWORD}
+spring.datasource.password=${JDBC_DATABASE_PASSWORD:btdevprod}
```
Comment (wolfs) [inline application/src/main/resources/application.properties:7]: I am not sure it makes sense to have a default password here.

**Review by wolfs — APPROVED** [review]
I think this looks good. When this works, then we could possibly try to upgrade to Spring Boot 3 as well, though I am not sure what changes there.

File: build-platform/build.gradle.kts (hunk 1 above)
Comment (hegyibalint) [reply build-platform/build.gradle.kts:20]: Very probable. As it's a bucket of dependencies that comes with Spring Boot, this is changing - most of the time additively.

File: application/src/main/resources/application.properties (hunk 4 above)
Comment (hegyibalint) [reply application/src/main/resources/application.properties:7]: It makes sense when you use suggested docker invocation.
https://github.com/gradle/bt-dev-prod-data-collector/blob/a2399b542f5e63aa25ce1e135c20c7e9e9d9ead9/persistence/build.gradle.kts#L22

**Comment by hegyibalint** [comment]
> When this works, then we could possibly try to upgrade to Spring Boot 3 as well, though I am not sure what changes there.

I found this doc, and I think we could attempt it, as I don't see anything directly impacting our app: https://spring.io/blog/2022/05/24/preparing-for-spring-boot-3-0

**Pull request merged**